gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -254,31 +250,67 @@ TypeCheckPattern::visit (HIR::StructPattern &pattern)
 	}
     }
 
-  if (named_fields.size () != variant->num_fields ())
+  // check the elements
+  if (adt->is_union ())
     {
-      std::map<std::string, bool> missing_names;
+      auto &struct_pattern_elems = pattern.get_struct_pattern_elems ();
+      if (struct_pattern_elems.get_struct_pattern_fields ().size () != 1)
+	rust_error_at (pattern.get_locus (),
+		       "union patterns does should have exactly one field");
 
-      // populate with all fields
-      for (auto &field : variant->get_fields ())
-	missing_names[field->get_name ()] = true;
-
-      // then eliminate with named_fields
-      for (auto &named : named_fields)
-	missing_names.erase (named);
-
-      // then get the list of missing names
-      size_t i = 0;
-      std::string missing_fields_str;
-      for (auto it = missing_names.begin (); it != missing_names.end (); it++)
+      else
+	{
+	  switch (struct_pattern_elems.get_struct_pattern_fields ()
+		    .at (0)
+		    ->get_item_type ())
+	    {
+	    case HIR::StructPatternField::ItemType::IDENT:
+	    case HIR::StructPatternField::ItemType::IDENT_PAT:
+	      break;
+	      default: {
+		auto first_elem
+		  = struct_pattern_elems.get_struct_pattern_fields ()
+		      .at (0)
+		      ->as_string ();
+		rust_error_at (pattern.get_locus (),
+			       "'%s' cannot be used in union patterns",
```

Comment:
```suggestion
			       "%qs cannot be used in union patterns",
```
to make use of GCC's quoting system

Reply:
thanks! I found it is documented in https://gcc.gnu.org/codingconventions.html  ☺️